### PR TITLE
proposing "warnings" for response meta

### PR DESCRIPTION
**Pull request by mbaudis**
Warnings seem like a much needed item to understand responses where the server wants to indicate information about potential problems with the response which are not outright failures. Especially useful for debugging aggregators.

### CODE REVIEW

**Comment by mbaudis** [comment]
@costero-e So I feel like changing `string` -> `object` (no further definitions, like `info` right now; or even using the `info` prototype. This would remove future hassle - one could design specific formats w/o breaking the spec. Agree? `info`?

**Comment by costero-e** [comment]
I don't know if it's better to put it as an object or not, what I see is that if we don't specify all the cases in which we want the beacon to return a warning, this will be very generic and I don't know if it will be useful then.

**Comment by redmitry** [comment]
> I don't know if it's better to put it as an object or not, what I see is that if we don't specify all the cases in which we want the beacon to return a warning, this will be very generic and I don't know if it will be useful then.

Unfortunately, the only place to report Beacon ERROR**~~S~~** is a dubious [BeaconErorr](https://github.com/ga4gh-beacon/beacon-v2/blob/2fe27f7da06ceed7fcac6c5a1aa496f31da3b115/framework/json/common/beaconCommonComponents.json#L17) object with an example "404" that has little sense to me. Beacon Network may use "warnings" as a placeholder for errors when we do have a response, but some beacons failed. Probably not best solution. I would extend beaconErrorResponse to use an array of BeaconErrors, but...

D.

**Comment by redmitry** [comment]
> @costero-e @redmitry I'm actually for keeping the format the way I've submitted (list of strings) since otherwise we run into read-out/interpretation problems, 

Agree.

I would also add the property location:
```json
"location" : {
    "type": string,
    "format": "json-pointer"
}
' ' 'json

As I mentioned before the error reporting is quite weak.
Json Schema recommends the format for **schema** errors reporting:
https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-01#section-12.4.2

D.

**Comment by costero-e** [comment]
@mbaudis The warning schema is more complete now, good job. On the other hand, I don't know if this is too technical. If this is oriented to users that don't have a lot of database or programming knowledge maybe these messages won't help them. But I guess every beacon developer can write the warning messages based on their needs so no problem.

@redmitry Can you provide an example of the location property, please?

**Comment by mbaudis** [comment]
@redmitry @costero-e `location`? Btw., have a look at the discussion there - maybe chime in? IMO it would be great to have the (geographic) `location` / `geoLocation` property in `service-info`. But a GeoLoaction object should anyway live somewhere & then be re-used across GA4GH standards; therefore I've had it for a while in {S}[B] ...

* https://github.com/ga4gh-discovery/ga4gh-service-info/issues/73 discussion in service-info
* {S}[B] GeoLocation  schema https://schemablocks.org/schema_pages/Progenetix/GeoLocation/

**Comment by redmitry** [comment]
> @redmitry Can you provide an example of the location property, please?

The idea is to use "location" which should locate the problem in json document (basically the input, but BN may use it for the outputs also)
.
In the json schema terms it is a "instanceLocation" for example; "/query/filters/0"
Json Schema parsers should also provide the "absoluteKeywordLocation" e.g. "https://raw.githubusercontent.com/ga4gh-beacon/beacon-v2/main/framework/json/requests/beaconRequestBody.json#/properties/query/properties/filters"

But as you said probably too technical. The idea is to provide some additional insight about the error location.
For instance, Beacon Network may use it to point invalid metadata provided by beacons.

D.

**Comment by costero-e** [comment]
@redmitry Ok, I see, thank you. It is interesting if we have all the possible errors listed in the Json schema.
@mbaudis geolocation? What would we be using it for? To map where we have all the beacons or the info?

**Comment by mbaudis** [comment]
Yes. You can direcly use a GeoJSON object to project onto a map, geofence/query, get stats ... See the linked discussion. Was reminded again by the handmade ELIXIR map of beacons. 
Now, while I wiuld prefer this as part of service-info (inherited by Beacon) we probably should demonstrate it in the network implementation (as additional parameter).

**Comment by jrambla** [comment]
Sorry for jumping in late in the discussion.
I would like to see the actual use cases we are trying to solve here.
The provided examples are not compelling to me:

- If a parameter could not be interpreted, an error should be returned or just a "false" response
- The discrepancy in the granularity between request and response could be detected looking at the response header, and that is the purpose for having them there

In summary, if "warnings" is like a "hack" we should discuss which are the actual issues we are trying to solve and if "warnings" solve them or we need different solutions for different use cases.
